CQRS/Domain/Repository/ProductRepository.cs

```diff
@@ -0,0 +1,59 @@
+﻿using CQRS.Domain.Entities;
+using CQRS.MongoDB.Base;
+
+using Microsoft.Extensions.Options;
+
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace CQRS.Domain.Repository;
+
+public class ProductRepository : IProductRepository
+{
+    private readonly IMongoCollection<Product> _product;
+
+    public ProductRepository(IOptions<MongoDBSettings> settingsOptions)
+    {
+        var client = new MongoClient(settingsOptions.Value.ConnectionString);
+        var userDatabase = client.GetDatabase(settingsOptions.Value.ProductDatabaseName);
+
+        _product = userDatabase.GetCollection<Product>("Warehouse");
```

Comment:
this is not product - these are products or productCollection, and why it points to Warehouse?